bitnami/external-dns/templates/deployment.yaml

```diff
@@ -396,7 +396,15 @@ spec:
                   name: {{ template "external-dns.secretName" . }}
                   key: rfc2136_tsig_secret
             {{- end }}
-            {{- if and (eq .Values.provider "pdns") (or .Values.pdns.apiKey .Values.pdns.secretName) }}
+            {{- if and (eq .Values.provider "pdns") .Values.pdns.apiKey }}
```

Comment:
I think you can use `.Values.pdns.apiKey` without adding this lines. Current changes take into account both cases, whether you provided with `.Values.pdns.apiKey` or `.Values.pdns.secretName`.

https://github.com/bitnami/charts/blob/5ed97f0c1ab49e7c7c3af81b888f41f50a7cfbab/bitnami/external-dns/templates/deployment.yaml#L399-L406

If no secret was provided, the Chart should be using `.Values.pdns.apiKey` by default:

https://github.com/bitnami/charts/blob/5ed97f0c1ab49e7c7c3af81b888f41f50a7cfbab/bitnami/external-dns/templates/_helpers.tpl#L152

What do you think?

Reply:
@joancafom do you mean  using `if  eq .Values.provider "pdns"`? As this chart will create the secret with pdns.apiKey when pdns.secretName is not provided in values.yaml.

Reply:
Yes, you can simplify to `if eq .Values.provider "pdns"` as the template function `external-dns.secretName` will retrieve the correct secret name, be it the user-provided one or the one created from the specified `.Values.pdns.apiKey`. Hence, as far as I can see, there is no need explicitly have these two condition:

```
...
{{- if and (eq .Values.provider "pdns") .Values.pdns.apiKey }}
...
{{- if and (eq .Values.provider "pdns") .Values.pdns.secretName }}
...
```